### PR TITLE
Add a small tolerance when testing pointer event positions

### LINKED ISSUE #6776: Tapping on map on Android doesn't always work after a Chrome fix
After we fixed [an old bug](https://crbug.com/632067) with PointerEvents in Chrome 58, an issue in OpenLayer's tap handling got exposed: a tap on Android doesn't always work.  More precisely, until Chrome 57 we used to (wrongly) suppress <code>pointermove</code> events in a small ("slop-suppression") region around the <code>pointerdown</code> point, which we fixed in Chrome 58.  After the fix, in OpenLayers tap doesn't work unless it's a very quick tap---the pointermoves within the old slop-region is perhaps triggering a pan followed by a "move end" event.

This is evident from the following examples:
- [box-selection.html](https://openlayers.org/en/latest/examples/box-selection.html)
- [moveend.html](https://openlayers.org/en/latest/examples/moveend.html)

I confirmed that it's not a Chrome bug through adding a hack <code>pointermove</code> listener through devtools that drops the move events close to the <code>pointerdown</code>.  I will close the [new Chrome bug](https://crbug.com/718540).

### PR DESCRIPTION
**Pull request by fredj**
fixes #6776 

demo: https://codepen.io/anon/pen/QvOPzp

### CODE REVIEW

**Comment by fredj** [comment]
@ischas can you test https://codepen.io/anon/pen/QvOPzp ?

**Comment by ischas** [comment]
@simonseyock ping

**Comment by simonseyock** [comment]
I am sorry, i can't test this. The problem does not occur on my phone.

File: src/ol/mapbrowsereventhandler.js [hunk 1]
```diff
@@ -281,8 +279,8 @@ ol.MapBrowserEventHandler.prototype.relayEvent_ = function(pointerEvent) {
  * @private
  */
 ol.MapBrowserEventHandler.prototype.isMoving_ = function(pointerEvent) {
-  return pointerEvent.clientX != this.down_.clientX ||
-      pointerEvent.clientY != this.down_.clientY;
+  return Math.abs(pointerEvent.clientX - this.down_.clientX) > 1 ||
+      Math.abs(pointerEvent.clientY != this.down_.clientY) > 1;
```
Comment (mustaqahmed) [inline src/ol/mapbrowsereventhandler.js]: Typo: the second condition should be `pointerEvent.clientY - this.down_.clientY`.

One question: Any chance the threshold `1` may be too large? The real offset that are causing the bug seems well below `1`. May be `0.5` or even `0.9` is a better alternative?

Comment (gastonmz) [inline src/ol/mapbrowsereventhandler.js]: I'm using 1 and works fine

**Comment by ischas** [comment]
For me it's nearly impossible to select a country in this demo:
http://openlayers.org/en/latest/examples/select-features.html

The linked CodePen works for me (but I guess @mustaqahmed is right with its review comment).

File: src/ol/mapbrowsereventhandler.js (hunk 1 above)
Comment (gastonmz) [inline src/ol/mapbrowsereventhandler.js]: I have problems handling the two fingers in order to make a pinch zoom...

Comment (fredj) [inline src/ol/mapbrowsereventhandler.js]: oups, thanks @mustaqahmed

Comment (fredj) [inline src/ol/mapbrowsereventhandler.js]: CodePen updated

**Comment by fredj** [comment]
I changed the threshold to 0.75 as suggested and updated the CodePen.

**Comment by ischas** [comment]
Now selecting is better than with master, but less easy as with the first commit.

**Comment by fredj** [comment]
@ischas back to 1px, CodePen updated

**Comment by ischas** [comment]
@simonseyock and I tested on my Samsung S5 with Android 6.0.1 and Chrome 58.

While with the select feature demo only round about every fifth to tenth touch is registered as touch, with the CodePen maybe every second touch is registered as touch. That's much better, but not good.

Please consider that the first commit behaved "better" in context of selection because of the bug and `isMoving_` always returning false.

**Comment by fredj** [comment]
I tested on a Samsung S6 and it's not working. I've added the `devicePixelRatio` into the mix

**Comment by ischas** [comment]
CodePen now works fine for me.

**Comment by fredj** [comment]
@ischas and are the pan, pinch zoom and rotate working as well?

**Comment by fredj** [comment]
@simonseyock can you also test on the phone that doesn't have the issue if everything continues to work as expected?

**Comment by ischas** [comment]
I Tested tap, double tap, pan, pinch zoom, rotate, everything's fine. Did I miss anything else?

@simonseyock is out of office until Friday, but maybe he'll find some time for testing in its spare time. ;-)

**Comment by fredj** [comment]
I'm tested with a Nexus 5x (doesn't have the issue) and it works as expected

**Comment by nagytech** [comment]
Tested Nexus 4, Android 5.1.1, Chrome 58.0.3029.83 - previously had the issue now working fine.

**Comment by fredj** [comment]
@openlayers/core thanks for any review

File: src/ol/mapbrowsereventhandler.js [hunk 2]
```diff
@@ -44,6 +45,12 @@ ol.MapBrowserEventHandler = function(map) {
   this.dragListenerKeys_ = [];
 
   /**
+   * @type {number}
+   * @private
+   */
+  this.moveTolerance_ = 1 * ol.has.DEVICE_PIXEL_RATIO;
```
Comment (ahocevar) [inline src/ol/mapbrowsereventhandler.js]: The `1 * ` is unnecessary.

**Comment by ischas** [comment]
Any chance for a patch release?

**Comment by mbj4668** [comment]
I have tested the CodePen with a Sony Z3 Compact, Android 6.0.1, Chrome 58.0.3029.83, and I still have problems selecting countries.  It works slightly better than http://openlayers.org/en/latest/examples/select-features.html, but not good.

I have also tested with a Sony Z4 Tablet, Android 7.0, Chrome 58.0.3029.83, and there the CodePen works fine, and the select-features example does not work.

**Comment by fredj** [comment]
What's the device pixel ratio of these devices ? see http://devicepixelratio.com/

**Comment by mbj4668** [comment]
On the Z3 compact it reports "2".

On the Z4 tablet it also reports "2".


/martin

2017-05-12 8:44 GMT+02:00 Frédéric Junod <notifications@github.com>:

**Comment by joaodiasafonso** [comment]
Hello.
I am facing this problem since Android webview was updated. If I uninstall latest version and stick to default one, it starts working.
Anyway, none of the workarounds/samples in this conversation worked for me. My device's pixel ratio is 4.
Thanks in advance. Regards,

**Comment by joaodiasafonso** [comment]
Need to make a correction because I was only checking in codepen and ol samples: after properly make @fredj update in library js, it began working as supposed. So, this has worked for me.

**Comment by jhwegener** [comment]
Nice PR, thanks @fredj !
Is there any chance to get a patch release for this?

**Comment by ahocevar** [comment]
We'll be releasing v4.2.0 in the next couple of days anyway.

**Pull request merged**